usdt-impl/src/no-linker.rs

```diff
@@ -60,27 +60,29 @@ fn compile_probe(
     let probe_rec = asm_rec(provider_name, probe.name(), Some(probe.types()));
     let pre_macro_block = TokenStream::new();
     let impl_block = quote! {
-        let mut is_enabled: u64;
-        // TODO can this block be option(pure)?
-        unsafe {
-            asm!(
-                "990:   clr rax",
-                #is_enabled_rec,
-                out("rax") is_enabled,
-                options(nomem, nostack, preserves_flags)
-            );
-        }
-
-        if is_enabled != 0 {
-            #unpacked_args
+        {
```

Comment:
this was important to keep the `is_enabled` variable out of the namespace, correct?

Reply:
Yeah, exactly, I want to pollute the caller's namespace as little as possible.